src/plugins/cesium/ngv-plugin-cesium-upload.ts

```diff
@@ -37,8 +37,42 @@ export class NgvPluginCesiumUpload extends LitElement {
   private uploadedModel: INGVCesiumModel | undefined;
 
   async upload(fileDetails: FileUploadDetails): Promise<void> {
-    const response = await fetch(fileDetails.url);
-    const arrayBuffer = await response.arrayBuffer();
+    let url = fileDetails.url;
+    let arrayBuffer =
+      fileDetails.file && (await fileDetails.file.arrayBuffer());
+    if (!arrayBuffer && url) {
+      const response = await fetch(url);
+      arrayBuffer = await response.arrayBuffer();
+    }
+    if (!arrayBuffer) {
+      return;
+    }
+
+    const decoder = new TextDecoder('utf-8');
+    const text = decoder.decode(arrayBuffer);
```

Comment:
Also, what happens when the loaded file is binary?
I guess the decode will fail.

Reply:
I tested, it works